apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/EventActivity.java

```diff
@@ -115,6 +121,20 @@ public void onClick(View v) {
          */
         mTransmissionTargets = EventActivityUtil.getAnalyticTransmissionTargetList(this);
 
+        String msaUserId = sSharedPreferences.getString(MSA_USER_ID_KEY, null);
+        if (msaUserId != null) {
+            long expirationDate = sSharedPreferences.getLong(MSA_EXPIRES_IN_KEY, 0);
+            Calendar calendar = Calendar.getInstance();
+            if (calendar.getTimeInMillis() > expirationDate) {
+                sSharedPreferences.edit().remove(MSA_EXPIRES_IN_KEY).apply();
+                sSharedPreferences.edit().remove(MSA_USER_ID_KEY).apply();
+            } else {
+                for(AnalyticsTransmissionTarget target : mTransmissionTargets) {
+                    target.getPropertyConfigurator().setUserId(msaUserId);
```

Comment:
The bug is about remembering the MSA token that is sent in headers, not the manual userId.
MSA token does not use an history (unlike MBaas), so the app could just restore the authentication callback with previous cached refresh token.